vertical-pod-autoscaler/pkg/recommender/checkpoint/checkpoint_writer.go

```diff
@@ -108,11 +108,9 @@ func (writer *checkpointWriter) StoreCheckpoints(ctx context.Context, now time.T
 			}
 			err = api_util.CreateOrUpdateVpaCheckpoint(writer.vpaCheckpointClient.VerticalPodAutoscalerCheckpoints(vpa.ID.Namespace), &vpaCheckpoint)
 			if err != nil {
-				klog.Errorf("Cannot save VPA %s/%s checkpoint for %s. Reason: %+v",
-					vpa.ID.Namespace, vpaCheckpoint.Spec.VPAObjectName, vpaCheckpoint.Spec.ContainerName, err)
+				klog.ErrorS(err, "Cannot save VPA checkpoint", "namespace", vpa.ID.Namespace, "vpa", vpaCheckpoint.Spec.VPAObjectName, "container", vpaCheckpoint.Spec.ContainerName)
```

Comment:
Let's simplify with `klog.KRef` and make clear that the reference is pointing to the VPA, not the checkpoint

```suggestion
				klog.ErrorS(err, "Cannot save checkpoint for VPA", klog.KRef(vpa.ID.Namespace, vpaCheckpoint.Spec.VPAObjectName), "container", vpaCheckpoint.Spec.ContainerName)
```